libexfat/io.c

```diff
@@ -20,29 +20,40 @@
 	51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 
+#define _GNU_SOURCE /* for fcntl.h fallocate */
+
 #include "exfat.h"
 #include <inttypes.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <stdarg.h>
+#include <stdio.h>
 #include <string.h>
 #include <errno.h>
+#ifdef HAVE_LINUX_FS_H
+#include <linux/fs.h>
+#endif
 #if defined(__APPLE__)
 #include <sys/disk.h>
 #elif defined(__OpenBSD__)
 #include <sys/param.h>
 #include <sys/disklabel.h>
 #include <sys/dkio.h>
+#endif
 #include <sys/ioctl.h>
-#elif __linux__
-#include <sys/mount.h>
+#ifdef __GLIBC__
+#include <sys/sysmacros.h>
```

Comment:
Is this for major()/minor()?

What will happen on non-GNU Linux-based system like Android?

Reply:
Yes, it's for `major()`/`minor()`. Bionic defines them in `sys/sysmacros.h` - and unline Glibc, doesn't have compat (with warnings, which is why I originally added the `#ifdef __GLIBC__`) in `sys/types.h`.

Switching to `AC_HEADER_MAJOR`, which should always do the right thing.